api/src/main/java/com/graphhopper/client/AbstractHopperClient.java

```diff
@@ -0,0 +1,122 @@
+package com.graphhopper.client;
+
+
+import com.graphhopper.GHResponse;
+import com.graphhopper.HopperClient;
+import com.graphhopper.HopperRequest;
+import com.graphhopper.HopperResponse;
+import com.graphhopper.bean.RouteBbox;
+import com.graphhopper.bean.RouteInstruction;
+import com.graphhopper.http.GraphHopperWeb;
+import com.graphhopper.http.WebHelper;
+import com.graphhopper.internal.HopperEngine;
+import com.graphhopper.internal.LazyPointList;
+import com.graphhopper.util.*;
+import com.graphhopper.util.shapes.BBox;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+// Please do not throw an error if hasError (with check everywhere)
+// Throws errors on errors is an anti-pattern
```

Comment:
The intend is to enforce the hasError check before calling any of the methods, so no anti pattern just making sure people really look into the errors. And what to return for the distance or points if there is an error?